spec/models/mixins/archived_mixin_spec.rb

```diff
@@ -0,0 +1,83 @@
+RSpec.describe ArchivedMixin do
+  let(:model_name) { :container }
+  let(:klass) { Container }
+
+  let(:timestamp) { Time.now.utc }
+  let(:active_model) { FactoryBot.create(model_name) }
+  let(:archived_model) { FactoryBot.create(model_name, :deleted_on => timestamp) }
+
+  describe ".archived" do
+    it "fetches archived records" do
+      active_model
+      archived_model
+
+      expect(klass.archived).to eq([archived_model])
+    end
+  end
+
+  describe ".active" do
+    it "fetches active records" do
+      active_model
+      archived_model
+
+      expect(klass.active).to eq([active_model])
+    end
+  end
+
+  describe ".active_for" do
```

Comment:
I think we need a few more cases (looking at my drawing that I shared the other day).

(created/deleted wording is relative to the time range in question (i.e. during means during the time range))

case | created | deleted | in range
--- | --- | --- | ---
a | before | before | ❌ 
b | before | during | ✅ 
c | before | after | ✅ 
d | before | not | ✅ 
e | during | during | ✅ 
f | during | after | ✅ 
g | during | not | ✅ 
h | after | after | ❌ 
i | after | not | ❌

Reply:
The problem: it is very tricky to define `created` across all container models.

@Fryguy Do you want to define different `active_on` scopes for each record?

Reply:
We don't need created for the first pass as discussed - I think h and i were hard for that reason.

Reply:
> @Fryguy Do you want to define different active_on scopes for each record?

Maybe?  Do have a list of the record types that are "different"?  Or possibly we can normalize a different way - perhaps @agrare has some thoughts here.